test/test_key_encode_decode.cpp

```diff
@@ -0,0 +1,365 @@
+                              static_cast<T>(0x090AULL));
+  do_encode_decode_order_test(static_cast<T>(0), static_cast<T>(1));
+  do_encode_decode_order_test(static_cast<T>(0x7FFFULL),
+                              static_cast<T>(0x8000ULL));
+  do_encode_decode_order_test(static_cast<T>(0xFFFEULL), static_cast<T>(~0ULL));
+}
+
+TEST(ARTKeyEncodeDecodeTest, Int16C00010) {
+  using T = std::int16_t;
+  const auto ONE = static_cast<T>(1);
+  // Check the encoder byte order.
+  const std::array<const std::byte, sizeof(T)> ikey{
+      static_cast<std::byte>(0x81), static_cast<std::byte>(0x02)};
+  do_encode_decode_test(static_cast<T>(0x0102LL), ikey);
+  // round-trip tests.
+  do_encode_decode_test(static_cast<T>(0));
+  do_encode_decode_test(static_cast<T>(0 + 1));
+  do_encode_decode_test(static_cast<T>(0 - 1));
+  do_encode_decode_test(std::numeric_limits<T>::min());
+  do_encode_decode_test(std::numeric_limits<T>::max());
+  do_encode_decode_test(std::numeric_limits<T>::min() + ONE);
+  do_encode_decode_test(std::numeric_limits<T>::max() - ONE);
+  // check lexicographic ordering for std::uint16_t pairs.
+  do_encode_decode_order_test(static_cast<T>(0), static_cast<T>(1));
+  do_encode_decode_order_test(static_cast<T>(5), static_cast<T>(7));
+  do_encode_decode_order_test(
+      std::numeric_limits<T>::min(),
+      static_cast<T>(std::numeric_limits<T>::min() + ONE));
+  do_encode_decode_order_test(
+      static_cast<T>(std::numeric_limits<T>::max() - ONE),
+      std::numeric_limits<T>::max());
+}
+
+TEST(ARTKeyEncodeDecodeTest, Uint32C00010) {
+  using T = std::uint32_t;
+  const auto ONE = static_cast<T>(1);
+  // Check the encoder byte order.
+  const std::array<const std::byte, sizeof(T)> ikey{
+      static_cast<std::byte>(0x01), static_cast<std::byte>(0x02),
+      static_cast<std::byte>(0x03), static_cast<std::byte>(0x04)};
+  do_encode_decode_test(static_cast<T>(0x01020304), ikey);
+  // round-trip tests.
+  do_encode_decode_test(static_cast<T>(0));
+  do_encode_decode_test(static_cast<T>(0 + 1));
+  do_encode_decode_test(static_cast<T>(0 - 1));
+  do_encode_decode_test(std::numeric_limits<T>::min());
+  do_encode_decode_test(std::numeric_limits<T>::max());
+  do_encode_decode_test(std::numeric_limits<T>::min() + ONE);
+  do_encode_decode_test(std::numeric_limits<T>::max() - ONE);
+  // check lexicographic ordering for std::uint32_t pairs.
+  do_encode_decode_order_test(static_cast<T>(0x01020304ULL),
+                              static_cast<T>(0x090A0B0CULL));
+  do_encode_decode_order_test(static_cast<T>(0), static_cast<T>(1));
+  do_encode_decode_order_test(static_cast<T>(0x7FFFFFFFULL),
+                              static_cast<T>(0x80000000ULL));
+  do_encode_decode_order_test(static_cast<T>(0xFFFFFFFEULL),
+                              static_cast<T>(~0ULL));
+}
+
+TEST(ARTKeyEncodeDecodeTest, Int32C00010) {
+  using T = std::int32_t;
+  const auto ONE = 1;  // by definition this is std::int32_t.
```

Comment:
Here and for other `ONE` instances: `constexpr`, lowercase name
(I don't see how it's by definition `int32_t`? It should be `int`)